### PR TITLE
Bug fixes, small feature enhancements and activity alert dialog

### PR DESCRIPTION
**Pull request by saleem-latif**
__Description:__
Here is a list of changes included in this PR.

- Show login prompt before repo-sync prompt.
- Added handling and guards agains null values of Project instance in different places inside the code base. This will help avoid null pointer exceptions.
- Explicitly set API timeout to 120 seconds for requests, also refactored client related code to add better error handling and make it cleaner.
- Included `added_at` parameter for new files so that correct time is used if user creates a file during the time period when user's plan has expired.
- Added a new dialog for showing user the prompt to review team activity.
- Added a new daemon to show alerts at 4:30 PM each day.
- Here is a screenshot of the new alert dialog
Screenshot 2023-01-02 at 2 04 28 PM

### CODE REVIEW

File: src/main/java/org/intellij/sdk/codesync/utils/DiffUtils.java [hunk 1]
```diff
@@ -41,6 +41,7 @@ public static void writeDiffToYml(
         }
         if (isNewFile) {
             data.put("is_new_file", true);
+            data.put("added_at", CommonUtils.getCurrentDatetime());
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/utils/DiffUtils.java:44]: I think this should be `created_at` of the file itself. PopulateBuffer may pick new file after some time (e.g. if the file was created outside of the IDE), thoughts?

File: src/main/java/org/intellij/sdk/codesync/Constants.java [hunk 2]
```diff
@@ -138,6 +146,12 @@ private Notification() {
         public static final String UPGRADE_ORG_PRICING_PLAN = "To continue, please sign your organization up for the Team plan.";
         public static final String TRY_PRO_FOR_FREE = "To continue, please sign up for a 30-day free trial of Pro.";
         public static final String TRY_ORG_PRO_FOR_FREE = "To continue, please sign your organization up for a 30-day free trial of the Team plan.";
+
+        // Team alert related notification messages
+        public static final String TEAM_ACTIVITY_ALERT_HEADER_MESSAGE = "CodeSync | Check your team's activity!";
+        public static final String TEAM_ACTIVITY_ALERT_MESSAGE = "Hey, check your team's activity!";
+        public static final String TEAM_ACTIVITY_ALERT_SECONDARY_MESSAGE =
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/Constants.java:153]: This is what we are using on VSCode: https://github.com/codesyncapp/codesync-vs/blob/master/src/constants.ts#L107

Let me know if I should update on the VSCode side.

**Review by Basit-qc — APPROVED** [review]
Thanks @saleem-latif , Looks great, includes fixes for many tickets 👍

File: src/main/java/org/intellij/sdk/codesync/utils/DiffUtils.java (hunk 1 above)
Comment (saleem-latif) [reply src/main/java/org/intellij/sdk/codesync/utils/DiffUtils.java:44]: I think we have the same behavior in VS Code as well. https://github.com/codesyncapp/codesync-vs/blob/3c5bbeeeb02c673bf9c6acb718668e066a404181/src/events/event_handler.ts#L64

**Pull request merged**